ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_composite_op.cpp

```diff
@@ -311,8 +325,14 @@ Tensor _div_no_nan_overload(const Tensor& input_a, float value, const std::optio
 }
 
 Tensor _div_no_nan(const Tensor& input_a, const Tensor& input_b, const std::optional<MemoryConfig>& output_mem_config) {
-    Tensor div_result = ttnn::div(input_a, input_b, false, std::nullopt, output_mem_config);
-    return ttnn::where(ttnn::eqz(input_b, output_mem_config), 0, div_result);
+    if (input_a.get_dtype() == DataType::FLOAT32 && input_b.get_dtype() == DataType::FLOAT32) {
+        // Not using SFPU div op here since inf/nan handling is not required
+        Tensor div_result = ttnn::multiply(input_a, ttnn::reciprocal(input_b), std::nullopt, output_mem_config);
+        return ttnn::where(ttnn::eqz(input_b, output_mem_config), 0.0f, div_result);
```

Comment:
Consider splitting this into a private function that checks the dtype and does the composite reciprocal & multiply or divide, that way you aren't copying so much code here.

Reply:
Sure. I am thinking of a way to do this at kernel level, but with a new api that exists only for such implementations not for user API.